### PR TITLE
rm check tools before cmds update tasks

### LINKED ISSUE #276: [Bug Report]: Build crashing on custom OpenOCD build (VSC-532)
**Pre Bug Report Checklist**
Before reporting any bug please make sure of these points.

- [x] Make sure you have searched for existing bugs and features request before you post an issue.
- [x] This is a bug report for the ESP-IDF Visual Studio Code extension and **not** an ESP-IDF bug report.
- [x] I've read the docs and found no information that could have helped solving the issue.

**Describe the bug**

I'm having similar symptoms as https://github.com/espressif/vscode-esp-idf-extension/issues/270, but my case is slightly different, I'll be thorough on this bug report...

After enabling `remote_bitbang` on OpenOCD (see PR https://github.com/espressif/openocd-esp32/pull/134), I cannot seem to be able to do anything with the extension anymore since the following error appears on any action (i.e Build):

```json
{
  "message": "Open On-Chip Debugger  v0.10.0-esp32-20201202-2-g7e342cad (2021-01-02-23:47)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n",
  "stack": "Error: Open On-Chip Debugger  v0.10.0-esp32-20201202-2-g7e342cad (2021-01-02-23:47)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n    at /Users/romanvg/.vscode-insiders/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7720\n    at ChildProcess.exithandler (child_process.js:303:7)\n    at ChildProcess.emit (events.js:315:20)\n    at ChildProcess.EventEmitter.emit (domain.js:483:12)\n    at maybeClose (internal/child_process.js:1021:16)\n    at Socket.<anonymous> (internal/child_process.js:443:11)\n    at Socket.emit (events.js:315:20)\n    at Socket.EventEmitter.emit (domain.js:483:12)\n    at Pipe.<anonymous> (net.js:674:12)",
  "level": "error",
  "timestamp": "2021-01-06T11:34:05.666Z"
}
```

The error log above occurs as well if having OpenOCD up and running... my best guess is that `idf_tools.py` version checking logic is interfering with the extension Build task/command?

**To Reproduce**

Install a slightly tweaked OpenOCD version (different than the one shipped with the extension).

**Expected behavior**

Everything working fine as it should? :)

**Screenshots**
On each Build task, the following error block appears in the extension's error log:

Skärmavbild 2021-01-06 kl  22 55 01

**Environment (please complete the following information):**

- OS Name & Version: MacOS X 10.15.2 (Catalina)
- VSCode Version: Insiders, latest
- ESP-IDF Version: `master`
- Python Version: 3.9.1

**Logs**

As shown above in the description.

**Additional context**

Please see PR https://github.com/espressif/openocd-esp32/pull/134 for more context. Also providing env vars I'm using:

```
  "idf.espIdfPath": "/Users/romanvg/dev/esp-idf",
  "idf.openOcdConfigs": [
    "interface/remote_bitbang.cfg",
    "target/esp32s2.cfg"
  ],
  "idf.customExtraVars": "{\"OPENOCD_SCRIPTS\":\"/Users/romanvg/.espressif/tools/openocd-esp32/v0.10.0-esp32-20201202/openocd-esp32/share/openocd/scripts\"}",
```

### LINKED ISSUE #277: [Bug Report]: Fix arguments order for monitor task in template tasks.json (VSC-533)
**Pre Bug Report Checklist**
Before reporting any bug please make sure of these points.

- [x] Make sure you have searched for existing bugs and features request before you post an issue.
- [x] This is a bug report for the ESP-IDF Visual Studio Code extension and **not** an ESP-IDF bug report.
- [x] I've read the docs and found no information that could have helped solving the issue.

**Describe the bug**
Current arguments order for windows command in monitor task cause the issues with flashing via running monitor - port option doesn't take into account. Especially when there is more than one available COM port.

**To Reproduce**
1. Connect additional COM port to Windows
2. Start monitor task
3. Try to build & flash via monitor (Ctrl+T Ctrl+F)

**Expected behavior**
Flash procedure uses correct port

**Environment (please complete the following information):**

- OS Name & Version: [Windows 10]
- VSCode Version: [1.52.1]
- ESP-IDF Version: 4.1]
- Python Version: [3.8.3]

**Additional context**
Easy to fix issue. Just change the following part of code:
```diff
"windows": {
-    "command": "${config:idf.pythonBinPathWin} ${config:idf.espIdfPathWin}\\tools\\idf.py monitor -p ${config:idf.portWin}"
+    "command": "${config:idf.pythonBinPathWin} ${config:idf.espIdfPathWin}\\tools\\idf.py -p ${config:idf.portWin} monitor"
},
```

Also, it would be nice if this task will use flash baud rate parameter.

### LINKED ISSUE #278: [Bug Report]: #274, for the monitor issue after 0.6.0 update, works only without the double quotes (VSC-534)
To get the internal terminal device monitor running again, i cloned the latest commit and rebuilded a new local vsix package for me.
It only worked without the additional double quotes inside the single back ticked (`) strings.

Win10
VSCode:
Version: 1.52.1 (system setup)
Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
Datum: 2020-12-16T16:34:46.910Z
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
Betriebssystem: Windows_NT x64 10.0.19042

```javascript
function getTxtCmd(variable: string, modifiedEnv: { [key: string]: string }) {
  const shellExecutable = path.basename(vscode.env.shell);
  switch (shellExecutable) {
    case "cmd.exe":
      return `set ${variable}=${modifiedEnv[variable]}`;
    case "powershell.exe":
    case "pwsh.exe":
      return `$Env:${variable}=${modifiedEnv[variable]}`;
    default:
      return `export ${variable}=${modifiedEnv[variable]}`;
  }
}
```

### LINKED ISSUE #280: [Bug Report]:  ESP-IDF Monitor device failed (VSC-535)
**Pre Bug Report Checklist**
Before reporting any bug please make sure of these points.

- [Y] Make sure you have searched for existing bugs and features request before you post an issue.
- [Y] This is a bug report for the ESP-IDF Visual Studio Code extension and **not** an ESP-IDF bug report.
- [Y] I've read the docs and found no information that could have helped solving the issue.

**Describe the bug**
My conclusion:
1. Monitoring device failed with version 0.6.0.

My observation:
1. Building, flashing and monitoring the device successfully with version 0.5.1
2. Building and flashing the device successfully with version 0.6.0.

My guess:
1. Python code is looking for an environment variable "IDF_PATH" when it is checking the IDF version while calling monitor the device

My approaches:
1. After version 0.6.0 installed, I ran through the configuration process again using "ESP-IDF: Configure ESP-IDF extension"
2. After the monitor fail, I manually entered `set IDF_PATH=C:\Users\aaa\Documents\GitHub\esp-idf` in the "ESP-IDF Monitor" terminal then rerun it again. It failed with a different reason `FileNotFoundError: [WinError 2] The system cannot find the file specified`. Stopped here.

**To Reproduce**
Steps to reproduce the behavior (including the full log).

1. Open VSCode
2. Open a folder contain a ESP-IDF project created with version 0.5.1
3. Click on the button "ESP-IDF Monitor Device"
4. Done
5. Please find the log in section Screenshots

**Expected behavior**
Print failure log

**Screenshots**
`
C:\Users\aaa\Documents\GitHub\bbb>c:\Users\aaa\Documents\ESP-IDF_Tools\v4.2\python_env\idf4.2_py3.8_env\Scripts\python.exe c:\Users\aaa\Documents\GitHub\esp-idf\tools\idf.py -p COM5 monitor
Traceback (most recent call last):
  File "c:\Users\aaa\Documents\GitHub\esp-idf\tools\idf.py", line 788, in <module>
    main()
  File "c:\Users\aaa\Documents\GitHub\esp-idf\tools\idf.py", line 704, in main
    checks_output = check_environment()
  File "c:\Users\aaa\Documents\GitHub\esp-idf\tools\idf.py", line 68, in check_environment
    print_idf_version()
  File "c:\Users\aaa\Documents\GitHub\esp-idf\tools\idf.py", line 117, in print_idf_version
    version = idf_version()
  File "c:\Users\aaa\Documents\GitHub\esp-idf\tools\idf_py_actions\tools.py", line 56, in idf_version
    "--git-dir=%s" % os.path.join(os.environ["IDF_PATH"], '.git'),
  File "C:\Users\aaa\AppData\Local\Programs\Python\Python38-32\lib\os.py", line 675, in __getitem__
    raise KeyError(key) from None
KeyError: 'IDF_PATH'
`

**Environment (please complete the following information):**

- OS Name & Version: Windows 10 (64-bit)
- VSCode Version: 1.52.1
- ESP-IDF Version: 4.2
- Python Version: 3.8.6

**Logs**
Please find the log in section Screenshots

**Extension Configuration settings**
Open Command Palette (F1) and type `Preferences: Open Settings (JSON)` or use `{CURRENT_PROJECT}/.vscode/settings.json` to get the following extension configuration settings value:
`
"idf.espIdfPathWin": "c:\\Users\\aaa\\Documents\\GitHub\\esp-idf",
    "idf.toolsPathWin": "c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2",
    "idf.pythonBinPathWin": "c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\python_env\\idf4.2_py3.8_env\\Scripts\\python.exe",
    "idf.customExtraPaths": "c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\python_env\\idf4.2_py3.8_env\\Scripts;C:\\Users\\aaa\\AppData\\Local\\Programs\\Python\\Python38-32;c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\tools\\xtensa-esp32-elf\\esp-2020r3-8.4.0\\xtensa-esp32-elf\\bin;c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\tools\\xtensa-esp32s2-elf\\esp-2020r3-8.4.0\\xtensa-esp32s2-elf\\bin;c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\tools\\esp32ulp-elf\\2.28.51-esp-20191205\\esp32ulp-elf-binutils\\bin;c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\tools\\esp32s2ulp-elf\\2.28.51-esp-20191205\\esp32s2ulp-elf-binutils\\bin;c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\tools\\cmake\\3.16.4\\bin;c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\tools\\openocd-esp32\\v0.10.0-esp32-20200709\\openocd-esp32\\bin;c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\tools\\ninja\\1.10.0;c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\tools\\idf-exe\\1.0.1;c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\tools\\ccache\\3.7;c:\\Users\\aaa\\Documents\\ESP-IDF_Tools\\v4.2\\tools\\dfu-util\\0.9\\dfu-util-0.9-win64",
    "idf.customExtraVars": "{\"OPENOCD_SCRIPTS\":\"c:\\\\Users\\\\aaa\\\\Documents\\\\ESP-IDF_Tools\\\\v4.2\\\\tools\\\\openocd-esp32\\\\v0.10.0-esp32-20200709/openocd-esp32/share/openocd/scripts\",\"IDF_CCACHE_ENABLE\":\"1\"}",
    "idf.showOnboardingOnInit": false,
    "idf.portWin": "COM5",
`

**Additional context**
Should be a fair easy fix
As build and flash is working, the new "Monitor" button just need to follow the same pattern to make it work.

### PR DESCRIPTION
**Pull request by brianignacio5**
Fix #280
Fix #278 
Fix #277
Fix #276 

Add env:PATH and reorder monitor args in tasks.json
Fix quotes on cmd.exe
Remove check esp-idf tools are valid from build flash commands.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -2451,7 +2435,7 @@ function getTxtCmd(variable: string, modifiedEnv: { [key: string]: string }) {
   const shellExecutable = path.basename(vscode.env.shell);
   switch (shellExecutable) {
     case "cmd.exe":
-      return `set ${variable}="${modifiedEnv[variable]}"`;
+      return `set "${variable}=${modifiedEnv[variable]}"`;
```
Comment (pwmb) [inline src/extension.ts:2438]: so if `modifiedEnv[variable] ` have spaced in them it will be fine?

Comment (brianignacio5) [reply src/extension.ts:2438]: yes the issue was the position of the quote. It was like this on v0.5.1 and before.

**Review by pwmb — APPROVED** [review]

**Comment by GOST-UA** [comment]
```
    if (canAccessCMake === "" && canAccessNinja === "") {
      throw new Error("CMake or Ninja executables not found");
    }
```
the condition doesn't match exception descriptions. I suggest it should be `||` instead of `&&`.

**Comment by GOST-UA** [comment]
I also propose to add `-b ${config:idf.flashBaudRate}` into monitor task to make its behavior similar to the rest of the tasks.

**Comment by brianignacio5** [comment]
@GOST-UA  For build we need both ninja-build and CMake. Tasks.json monitor task can be modified by user after project creation and the monitor baud rate can be configured in the SDK Configuration Editor with CONFIG_ESPTOOLPY_MONITOR_BAUD.

**Pull request merged**